misc-scripts/xref_mapping/XrefParser/RefSeqGPFFParser.pm

```diff
@@ -164,30 +164,48 @@ sub create_xrefs {
 
   local $/ = "\/\/\n";
 
-  my $type;
-  if ($file =~ /protein/) {
+  my $type = $self->type_from_file($file);
+  return unless $type;
 
-    $type = 'peptide';
-
-  } elsif ($file =~ /rna/) {
-
-    $type = 'dna';
+  while ( $_ = $refseq_io->getline() ) {
```

Comment:
Nice one! In the previous code $_ was referred to in a method body , even though the code also did  `my $entry = $_;`. I only saw it because it stopped referring to anything when I applied your remark and I got a warning. So I think the behaviour is still the same, but more futureproof when the previous version will start doing something else when $_ gets clobbered.